### PR TITLE
feat: project variables create

### PR DESCRIPTION
**Pull request by benPearce1**
feat: project variables update
feat: project variables list
feat: project variables view
feat: project variables delete

~~depends on https://github.com/OctopusDeploy/go-octopusdeploy/pull/181~~

list
image

view variable with multiple values
image

view variable with multiple scopes
image

update variable, replace scope
image

update, set new value
image

update, remove all scopes
image

updating variable by name with multiple values
image

create, prompted variable, no default value
image

image

delete
image

### CODE REVIEW

File: pkg/cmd/project/variables/create/create.go [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+			for {
+				var value string
+
+				if err := opts.Ask(&survey.Input{
+					Message: "Enter a selection option value (enter blank to end)",
+				}, &value, survey.WithValidator(survey.MaxLength(200))); err != nil {
+					return err
+				}
+
+				if value == "" {
```
Comment (domenicsim1) [inline pkg/cmd/project/variables/create/create.go:281]: value here should strip white space incase user hit space before enter.

File: pkg/cmd/project/variables/create/create.go [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+				}, &description, survey.WithValidator(survey.ComposeValidators(survey.Required))); err != nil {
+					return err
+				}
+
+				opts.PromptSelectOptions.Value = append(opts.PromptSelectOptions.Value, fmt.Sprintf("%s|%s", value, description))
+			}
+		}
+
+		if !opts.PromptRequired.Value {
+			opts.Ask(&survey.Confirm{
```
Comment (domenicsim1) [inline pkg/cmd/project/variables/create/create.go:297]: very unlikely to happen but err assignment and check is missing here.

**Review by domenicsim1 — APPROVED** [review]
Some little things, but overall looks good.

File: pkg/cmd/project/variables/create/create.go (hunk 2 above)
Comment (benPearce1) [reply pkg/cmd/project/variables/create/create.go]: good catch

**Pull request merged**